Library/Homebrew/upgrade.rb

```diff
@@ -81,8 +81,9 @@ def upgrade_formulae(
                 nil
               end
               next true unless installed_version
+              next true if hash["revision"].nil? # Tabs prior to 4.1.18 did not have this
```

Comment:
As mentioned above: I think this is too conservative. We should do something like e.g. looking at the formula and seeing if it currently has any `revision`.